src/piqtree2/_app/__init__.py

```diff
@@ -2,35 +2,49 @@
 
 import cogent3
 import cogent3.app.typing as c3_types
-from cogent3.app.composable import define_app
+from cogent3.app import composable
 from cogent3.util.misc import extend_docstring_from
 
 from piqtree2 import TreeGenMode, build_tree, fit_tree, nj_tree, random_trees
-from piqtree2.model import Model
+from piqtree2.model import (
+    Model,
+    available_freq_type,
+    available_models,
+    available_rate_type,
+)
 
 
-@define_app
-@extend_docstring_from(build_tree)
-def piqtree_phylo(
-    aln: cogent3.Alignment | cogent3.ArrayAlignment,
-    model: Model,
-    rand_seed: int | None = None,
-) -> cogent3.PhyloNode | cogent3.app.typing.SerialisableType:
-    return build_tree(aln, model, rand_seed)
+@composable.define_app
+class piqtree_phylo:
+    @extend_docstring_from(build_tree)
+    def __init__(
+        self,
+        model: str,
+        rand_seed: int | None = None,
+    ) -> None:
+        self._model = Model(model)
+        self._rand_seed = rand_seed
+
+    def main(
+        self,
+        aln: cogent3.Alignment | cogent3.ArrayAlignment,
+    ) -> cogent3.PhyloNode | cogent3.app.typing.SerialisableType:
+        return build_tree(aln, self._model, self._rand_seed)
 
 
-@define_app
+@composable.define_app
 @extend_docstring_from(fit_tree)
 def piqtree_fit(
     aln: cogent3.Alignment | cogent3.ArrayAlignment,
     tree: cogent3.PhyloNode,
     model: Model,
```

Comment:
Is this meant to be `str` now? (as in the new `piqtree_phylo`)

Reply:
yes, should mirror `piqtree_phylo`. I also think this should become a class. The problem with the function based apps is that the `app_help()` is not working well, does not produce a code snippet that can be copied and pasted to be a functional app/